finality-provider/service/chain_poller.go

```diff
@@ -97,6 +97,10 @@ func (cp *ChainPoller) Start(startHeight uint64) error {
 }
 
 func (cp *ChainPoller) Stop() error {
+	if !cp.IsRunning() {
```

Comment:
without this, `TestOpFpNoVotingPower` will fail b/c it's trying to stop a poller that's not started

Reply:
isn't this the same as `!cp.isStarted.Swap(false)`? It's just nil is returned instead of an error

Reply:
yes, now `TestOpFpNoVotingPower` will start the poller, it can be removed.